src/dynamics/mpas/driver/dyn_mpas_subdriver.F90

```diff
@@ -485,4 +705,1520 @@ subroutine dyn_mpas_init_phase2(self, pio_iosystem)
+                select case (mpas_pool_field_info % ndims)
+                    case (1)
+                        call mpas_pool_get_field(self % domain_ptr % blocklist % allfields, &
+                            trim(adjustl(field_name)), field_1d_real, timelevel=1)
+
+                        if (.not. associated(field_1d_real)) then
+                            call self % model_error('Failed to find field', subname, __LINE__)
+                        end if
+
+                        call mpas_dmpar_exch_halo_field(field_1d_real)
+
+                        nullify(field_1d_real)
+                    case (2)
+                        call mpas_pool_get_field(self % domain_ptr % blocklist % allfields, &
+                            trim(adjustl(field_name)), field_2d_real, timelevel=1)
+
+                        if (.not. associated(field_2d_real)) then
+                            call self % model_error('Failed to find field', subname, __LINE__)
+                        end if
+
+                        call mpas_dmpar_exch_halo_field(field_2d_real)
+
+                        nullify(field_2d_real)
+                    case (3)
+                        call mpas_pool_get_field(self % domain_ptr % blocklist % allfields, &
+                            trim(adjustl(field_name)), field_3d_real, timelevel=1)
+
+                        if (.not. associated(field_3d_real)) then
+                            call self % model_error('Failed to find field', subname, __LINE__)
+                        end if
+
+                        call mpas_dmpar_exch_halo_field(field_3d_real)
+
+                        nullify(field_3d_real)
+                    case (4)
+                        call mpas_pool_get_field(self % domain_ptr % blocklist % allfields, &
+                            trim(adjustl(field_name)), field_4d_real, timelevel=1)
+
+                        if (.not. associated(field_4d_real)) then
+                            call self % model_error('Failed to find field', subname, __LINE__)
+                        end if
+
+                        call mpas_dmpar_exch_halo_field(field_4d_real)
+
+                        nullify(field_4d_real)
+                    case (5)
+                        call mpas_pool_get_field(self % domain_ptr % blocklist % allfields, &
+                            trim(adjustl(field_name)), field_5d_real, timelevel=1)
+
+                        if (.not. associated(field_5d_real)) then
+                            call self % model_error('Failed to find field', subname, __LINE__)
+                        end if
+
+                        call mpas_dmpar_exch_halo_field(field_5d_real)
+
+                        nullify(field_5d_real)
+                    case default
+                        call self % model_error('Unsupported field rank', subname, __LINE__)
+                end select
+            case default
+                call self % model_error('Unsupported field type', subname, __LINE__)
```

Comment:
Here the `select case` construct tests if `mpas_pool_field_info % fieldtype` equals to either of the predefined **integer constants**, `mpas_pool_integer` and `mpas_pool_real`. If `mpas_pool_field_info % fieldtype` equals to neither of them, it is unclear what type it actually is (because it is undefined...). So, we cannot print a type string here. Instead, we can print what types are supported, like below in https://github.com/ESCOMP/CAM-SIMA/pull/257/commits/35dd726a585e35d089bea6736bd2dadada3327cb:

https://github.com/ESCOMP/CAM-SIMA/blob/35dd726a585e35d089bea6736bd2dadada3327cb/src/dynamics/mpas/driver/dyn_mpas_subdriver.F90#L1569-L1570

Reply:
Right!  Sorry, I guess I have been spending too much time working in python recently.  I think just listing the allowed types in the error message is fine here.  Thanks!